### ISSUE TITLE
The hash constructor fails where the JSON file it was built from does not

### ISSUE THREAD

**Issue by JJ** [jonathanstowe/META6#27]
Take any distribution, use META6.new( json => $json); then META6.new(  | from-json( $json )). It will very definitely fail, starting with versions and meta versions, which need Version objects.
This is probably fair, but it should be either well documented or simply have a default behavior that works, and an alternative constructor that indicates clearly that it needs specifics, probably with named, typed arguments instead of the raw Hash.

**Comment by jonathanstowe** [comment]
Yeah, this is down to the way that `JSON::Unmarshal` and `JSON::Class` work,  it's tricky to fix without reimplementing both things.
